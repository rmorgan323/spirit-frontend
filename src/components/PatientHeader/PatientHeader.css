/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 16px;
  font-family: "Lato", sans-serif;
  outline: none;
  color: #6d6e71; }

h1, h2, h3, h4, h5, h6 {
  font-family: "Josefin Sans", sans-serif;
  line-height: 1.3; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 30px; }

h3 {
  font-size: 24px; }

h4 {
  font-size: 18px; }

h5 {
  font-size: 14px; }

h6 {
  font-size: 10px; }

p {
  font-size: 14px;
  line-height: 1.3;
  font-weight: 900; }

::placeholder {
  opacity: .5;
  text-align: center; }

/***  MIXINS  ***/
.PatientHeader {
  background-color: #5E378C; }
  .PatientHeader .patient-header-holder {
    padding: 10px 20px 8px; }
  .PatientHeader h4 {
    margin: 0;
    color: #9E87BA;
    font-size: 14px; }
  .PatientHeader .patient-header-span {
    color: #EAE5F0;
    font-size: 18px;
    margin: 0 30px 0 10px; }
