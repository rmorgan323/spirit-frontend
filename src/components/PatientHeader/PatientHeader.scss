@import '../../variables.scss';

.PatientHeader {
  background-color: $purple2;

  .patient-header-holder {
    padding: 10px 20px 8px;
  }

  h4 {
    margin: 0;
    color: $purple4;
    font-size: 14px;
  }

  .patient-header-span {
    color: $purple6;
    font-size: 18px;
    margin: 0 30px 0 10px;
  }

}