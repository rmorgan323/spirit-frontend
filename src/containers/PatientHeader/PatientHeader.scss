@import '../../variables.scss';

.PatientHeader {
  background-color: $purple6;

  .patient-header-holder {
    padding: 10px 20px 8px;
  }

  h4 {
    margin: 0;
    color: $purple4;
    font-size: 14px;
  }

  .patient-header-span {
    color: $purple1;
    font-size: 18px;
    margin: 0 30px 0 10px;
  }

}