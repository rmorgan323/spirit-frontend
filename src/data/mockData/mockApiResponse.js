const mockApiResponse = {
  status: 'success'
};

export default mockApiResponse;