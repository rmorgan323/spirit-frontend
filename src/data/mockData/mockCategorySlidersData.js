/*eslint-disable camelcase*/

export const mockTitle = 'Capacities';

export const mockSlidersArray = [
  'Attention',
  'Action',
  'Affect',
  'Autonomic',
  'Arousability'
];

export const mockDatabaseNamesArray = [
  'mod_2_attention',
  'mod_2_action',
  'mod_2_affect',
  'mod_2_autonomic',
  'mod_2_arousability'
];

export const mockReminderAsterisk = [null, '*', '*', '*', '*'];

export const mockReminder = '* If selected, remember to make a hypothesized arousal chart.';

export const mockSliderTitle = 'Attention';

export const mockDatabaseName = 'mod_2_attention';

