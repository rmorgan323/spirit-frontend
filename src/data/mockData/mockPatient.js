/*eslint-disable camelcase*/

export const mockPatient = {
  id: 5,
  abstracted_name: "INTrm5",
  clinic_name: "I Need Therapy",
  ot_id: 1,
  created_at: "2018-02-20T20:07:54.509Z",
  updated_at: "2018-02-20T20:07:54.509Z"
};
