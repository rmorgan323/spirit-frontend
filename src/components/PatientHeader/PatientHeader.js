import React from 'react'
import './PatientHeader.css';
import { connect } from 'react-redux';
import moment from 'moment';

const PatientHeader = (props) => {

  let display;

  if (props.currentPatient.id !== undefined) {
    const patientDisplay = props.currentPatient.abstracted_name;
    const sinceDisplay = moment(props.currentPatient.created_at).format('LL')
    display = (
      <div className="patient-header-holder">
      <h4>PATIENT:<span className="patient-header-span">{patientDisplay}</span></h4>
      </div>
    )
  }

  return (
    <div className="PatientHeader">
      {display}
    </div>
  )
}

const mapStateToProps = store => ({
  currentPatient: store.currentPatient
})

export default connect(mapStateToProps, null)(PatientHeader);
