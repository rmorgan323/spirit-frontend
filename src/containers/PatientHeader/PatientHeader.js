import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import './PatientHeader.css';
// import moment from 'moment';

export const PatientHeader = props => {
  let display;

  if (props.currentPatient.id !== undefined) {
    const patientDisplay = props.currentPatient.abstracted_name;
    // const sinceDisplay = moment(props.currentPatient.created_at).format('LL')
    display = (
      <div className="patient-header-holder">
        <h4>
          PATIENT:<span className="patient-header-span">{patientDisplay}</span>
        </h4>
      </div>
    );
  }

  return <div className="PatientHeader">{display}</div>;
};

export const mapStateToProps = store => ({
  currentPatient: store.currentPatient
});

export default connect(mapStateToProps, null)(PatientHeader);

PatientHeader.propTypes = {
  currentPatient: PropTypes.object
};
