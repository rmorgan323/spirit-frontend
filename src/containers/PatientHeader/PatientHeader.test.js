import React from 'react';
import { PatientHeader, mapStateToProps } from './PatientHeader';
import { shallow } from 'enzyme';
import { mockPatient } from '../../data/mockData/mockPatient';

describe('Patient Header tests', () => {
  let currentPatient;
  let renderedPatientHeader;

  beforeEach(() => {
    currentPatient = mockPatient;

    renderedPatientHeader = shallow(
      <PatientHeader currentPatient={currentPatient} />
    );
  });

  it('Should match the snapshot', () => {
    expect(renderedPatientHeader).toMatchSnapshot();
  });
});

describe('mapStateToProps tests', () => {
  it('Should pull a currentPatient from store', () => {
    const currentPatient = mockPatient;
    const mockStore = { currentPatient };
    const result = mapStateToProps(mockStore);

    expect(result.currentPatient).toEqual(mockStore.currentPatient);
  });
});
