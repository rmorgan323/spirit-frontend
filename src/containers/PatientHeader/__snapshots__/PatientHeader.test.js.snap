// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Patient Header tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentPatient={
      Object {
        "abstracted_name": "INTrm5",
        "clinic_name": "I Need Therapy",
        "created_at": "2018-02-20T20:07:54.509Z",
        "id": 5,
        "ot_id": 1,
        "updated_at": "2018-02-20T20:07:54.509Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="patient-header-holder"
      >
        <h4>
          PATIENT:
          <span
            className="patient-header-span"
          >
            INTrm5
          </span>
        </h4>
      </div>,
      "className": "PatientHeader",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4>
          PATIENT:
          <span
            className="patient-header-span"
          >
            INTrm5
          </span>
        </h4>,
        "className": "patient-header-holder",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "PATIENT:",
            <span
              className="patient-header-span"
            >
              INTrm5
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "PATIENT:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "INTrm5",
              "className": "patient-header-span",
            },
            "ref": null,
            "rendered": "INTrm5",
            "type": "span",
          },
        ],
        "type": "h4",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="patient-header-holder"
        >
          <h4>
            PATIENT:
            <span
              className="patient-header-span"
            >
              INTrm5
            </span>
          </h4>
        </div>,
        "className": "PatientHeader",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4>
            PATIENT:
            <span
              className="patient-header-span"
            >
              INTrm5
            </span>
          </h4>,
          "className": "patient-header-holder",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "PATIENT:",
              <span
                className="patient-header-span"
              >
                INTrm5
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "PATIENT:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "INTrm5",
                "className": "patient-header-span",
              },
              "ref": null,
              "rendered": "INTrm5",
              "type": "span",
            },
          ],
          "type": "h4",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
